yt_dlp/extractor/kenh14.py

```diff
@@ -0,0 +1,150 @@
+            'description': 'md5:de86aa22e143e2b277bce8ec9c6f17dc',
+            'thumbnail': r're:^https?://videothumbs\.mediacdn\.vn/.*\.jpg$',
+            'tags': [],
+            'upload_date': '20220517',
+            'view_count': int,
+            'duration': 70.04,
+            'timestamp': 1652766021,
+        },
+    }, {
+        'url': 'https://video.kenh14.vn/0-344740.chn',
+        'md5': 'b843495d5e728142c8870c09b46df2a9',
+        'info_dict': {
+            'id': '344740',
+            'ext': 'mov',
+            'title': 'Kỳ Duyên đầy căng thẳng trong buổi ra quân đi Miss Universe, nghi thức tuyên thuệ lần đầu xuất hiện gây nhiều tranh cãi',
+            'description': 'md5:2a2dbb4a7397169fb21ee68f09160497',
+            'thumbnail': r're:^https?://kenh14cdn\.com/.*\.jpg$',
+            'tags': ['kỳ duyên', 'Kỳ Duyên tuyên thuệ', 'miss universe'],
+            'uploader': 'Quang Vũ',
+            'upload_date': '20241024',
+            'view_count': int,
+            'duration': 198.88,
+            'timestamp': 1729741590,
+        },
+        'expected_warnings': [
+            'Failed to download m3u8 information: HTTP Error 404: NOT FOUND',
+        ],
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        attrs = extract_attributes(get_element_html_by_attribute('type', 'VideoStream', webpage) or '')
+        direct_url = attrs['data-vid']
+
+        metadata = self._download_json(
+            'https://api.kinghub.vn/video/api/v1/detailVideoByGet?FileName={}'.format(
+                remove_start(direct_url, 'kenh14cdn.com/')), video_id, fatal=False)
+
+        return {
+            'id': video_id,
+            'title': (
+                strip_or_none(metadata.get('title'))
+                or clean_html(self._og_search_title(webpage))
+                or clean_html(get_element_by_class('vdbw-title', webpage))),
+            'formats': [
+                {'url': f'https://{direct_url}', 'format_id': 'http'},
+                *self._extract_m3u8_formats(f'https://{direct_url}/master.m3u8', video_id, fatal=False),
+            ],
+            'duration': parse_duration(metadata.get('duration')),
+            'description': (
+                clean_html(self._og_search_description(webpage))
+                or clean_html(get_element_by_class('vdbw-sapo', webpage))),
+            'thumbnail': (self._og_search_thumbnail(webpage) or attrs.get('data-thumb')),
+            'uploader': strip_or_none(metadata.get('author')),
+            'timestamp': parse_iso8601(metadata.get('uploadtime'), delimiter=' '),
+            'view_count': int_or_none(metadata.get('views')),
+            'tags': traverse_obj(self._html_search_meta('keywords', webpage), (
+                {lambda x: x.split(';')}, lambda _, v: v, {str_or_none})),
```

Comment:
might need to merge master again and you can do this
```suggestion
                {lambda x: x.split(';')}, filter)),
```

Reply:
Which `filter`? I searched in the newest code but can only find `def filter_dict` ...

Reply:
`filter` is a builtin python function. In traversal it removes failed values, so `{}` and `None`. See also the updated docstring for `traverse_obj`

Reply:
Thank you! And I didn't realize the docstring has been updated. my bad. Let me commit the subsequent patch.